### ISSUE TITLE
Improve HoC connectToStores Docs

### ISSUE THREAD

**Issue by burabure** [acdlite/flummox#136]
I think that the connectToStores HoC pattern is a huge plus for Flummox, and that the actual docs don't showcase how awesome it is (Easy, intuitive Relay-like data fetching from stores).

I'd be happy to send  a PR with an updated Quick start, examples and a "Why HoC > FluxComponent > fluxMixin". Also I could introduce

``` 
contextTypes: {
    flux: React.PropTypes.object
  },
```

because without it HoC becomes very clunky.

@acdlite basically I just want to check if this is the direction that you want on the guide/docs?

**Comment by acdlite** [comment]
Yes to all, that would be a huge help

**Comment by burabure** [comment]
just checking: right now the HoC needs a higher FlummoxComponent to pass the flux object to the context right?. or is there a way to pass the flux object to the HoC?

**Comment by acdlite** [comment]
Correct, currently no way to pass Flux directly to the HoC

**Comment by burabure** [comment]
Slightly unrelated: are there plans to add this behavior to the HoC?, should I open an issue?

**Comment by acdlite** [comment]
Yeah, eventually. Also want to add [`injectActions`](https://github.com/acdlite/flummox/issues/132), but I'm not sure what the best API for that is yet.

**Comment by burabure** [comment]
been working on the docs!. Quickstart and React integration are ready. i'll be doing "Why HoC > FluxComponent > fluxMixin" next week!

https://github.com/acdlite/flummox/compare/master...burabure:master

**Comment by barrystaes** [comment]
:+1: Documentation on what connectToStores() does (notably what do its parameters represent) would be most welcome.

Once i hit the point where "it works but i dont know why" i figured out what does what by intentionally breaking stuff, but its not exactly evident.

**Comment by dozoisch** [comment]
:+1:

**Comment by akofman** [comment]
Thanks a lot for these docs @burabure ! It helped me a loooot.
If I could make a comment on this part  : (https://github.com/burabure/flummox/blob/master/docs/docs/guides/react-integration.md)

```
// Pass an object of store keys mapped to getter functions
MyComponent = connectToStores(MyComponent, {  
  posts: store => ({
    postBody: store.getPostBody(this.props.post.id),
  }),
  comments: store => ({
    comments: store.getCommentsForPost(this.props.post.id),
  })
});
```

Depends on your module loading system, if you're using io.js, `this` will be ok but in certain cases like with Babel, `this` could be undefined.
=> https://github.com/babel/babel/issues/562

I think it could be helpful for those like me to include this point in the documentation. Thoughts ?

**Comment by burabure** [comment]
thanks @akofman, happy to help =)

about the issue: I think you're right. with ES6 arrow functions ```this``` would point to the outside scope instead of connectToStores context. thanks!

i'll fix it right away!

**Comment by acdlite** [comment]
Store state getters are not bound to the component instance, as of v3.0, so even without an arrow function, `this.props` will always be undefined. The way to do this now is to use the `props` parameter:

```js
// Pass an object of store keys mapped to getter functions
MyComponent = connectToStores(MyComponent, {  
  posts: (store, props) => ({
    postBody: store.getPostBody(props.post.id),
  }),
  comments: (store, props) => ({
    comments: store.getCommentsForPost(props.post.id),
  })
});
```

**Comment by acdlite** [comment]
Oops, I hate that button :D

**Comment by burabure** [comment]
thank you @acdlite. is that behaviour documented?, I think there's something about them in the upgrade guide, but it seems everything else is outdated. 

I'll add it to the component, HoC and mixin just in case =)

**Comment by acdlite** [comment]
I don't think it's explicitly documented anywhere, just reflected in the examples.

Thanks :)

**Comment by burabure** [comment]
just checking: ```this.props``` on the fluxcomponent would also be undefined since 3.0 right?

```js
class BlogPostPage extends React.Component {
  render() {
    <div>
      <SiteNavigation />
      <MainContentArea>
        <FluxComponent connectToStores={{
          posts: store => ({
            post: store.getPost(this.props.postId),
          })
        }}>
          <BlogPost />
        </FluxComponent>
      </MainContentArea>
      <SiteSidebar />
      <SiteFooter />
    </div>
  }
}
```

**Comment by acdlite** [comment]
In that case it refers to the owner (BlogPostPage), which is probably what you want.

Double wrapping is correct because you're passing an object. The first set of braces is for embedding a JavaScript value in JSX; the second set is the object itself. It's the same concept as when you pass an inline object to the `style` props.

**Comment by burabure** [comment]
stateGetter's second param (props) in the docs now.

https://github.com/burabure/flummox/commit/145f977adf8f6bd2028cd6231321cdbf882a33bd

@acdlite if you have a couple minutes for a quick check, in case I messed up some detail, it would be awesome =)

**Comment by burabure** [comment]
typo fixed and proof read. ready to pull in =)

**Comment by acdlite** [comment]
Thanks! I'll try to get this merged in tonight. Sorry I've been a bit less responsive lately — I accepted a new job in a new city, so I've been really busy :)

**Comment by burabure** [comment]
hope the new job and the moving stuff are getting less crazy ;D

Let me know if I can help with anything to make this PR easier to pull

**Comment by johanneslumpe** [comment]
@acdlite we should get this merged I think :)

**Comment by swordsreversed** [comment]
+1 this would be helpful to have merged. I cloned vesparny's repo and it uses the HoC, and with no documentation it was pretty confusing.

**Comment by vicentedealencar** [comment]
+1

**Comment by mmerickel** [comment]
While the function itself is anonymous - now that `connectToStores` accepts actions I'd recommend renaming its usage in the docs to `connectToFlux` or something similar.
